src/Symfony/Component/Lock/Store/MongoDbStore.php

```diff
@@ -0,0 +1,276 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\Store;
+
+use Symfony\Component\Lock\Exception\InvalidArgumentException;
+use Symfony\Component\Lock\Exception\LockConflictedException;
+use Symfony\Component\Lock\Exception\LockExpiredException;
+use Symfony\Component\Lock\Exception\LockStorageException;
+use Symfony\Component\Lock\Exception\NotSupportedException;
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\StoreInterface;
+
+/**
+ * MongoDbStore is a StoreInterface implementation using MongoDB as a storage
+ * engine.
+ *
+ * @author Joe Bennett <joe@assimtech.com>
+ */
+class MongoDbStore implements StoreInterface
+{
+    private $mongo;
+    private $options;
+    private $collection;
+
+    /**
+     * @param \MongoDB\Client $mongo
```

Comment:
Is there a way to pass a dns for lazy connection? See PdoTrait from cache components?

Reply:
The only way to avoid having to pass the database name as an options is to pass a `\MongoDB\Database` in `MongoDbStore::__construct` instead of a `\MongoDB\Client`.

A mongodb DSN does not contain a database to `use`. https://docs.mongodb.com/manual/reference/connection-string/

The `/database` component there is the database for authentication (unrelated to the database containing the data).

If I were to allow either a `\MongoDB\Database` or a `\MongoDB\Client`; the `database` option would then have to become mandatory only if a `\MongoDB\Client` is given.

Reply:
A `MongoDB\Client` instance is already lazy. I can turn off my database and still:
```php
$client = new \MongoDB\Client();
$store = new \Symfony\Component\Lock\Store\MongoDbStore($client, [
    'database' => 'test',
]);
```

only if I add something like:

```php
$store->createTTLIndex();
```

does it throw an exception.